### PR TITLE
Check cgroup controllers existence

### PR DESCRIPTION
**Pull request by lizf-os**
We already have checks for MemoryLimit, CpuPeriod and CpuQuota, but there's no check for Cpuset, CpuShares and BlkioWeight. As a result docker will fail to start containers if those settings are specified but the underlying kernel doesn't support them.

### CODE REVIEW

**Comment by cpuguy83** [comment]
This needs a rebase

File: pkg/sysinfo/sysinfo_linux.go [hunk 1]
```diff
@@ -40,7 +40,13 @@ func New(quiet bool) *SysInfo {
 			logrus.Warnf("%v", err)
 		}
 	} else {
-		_, err := ioutil.ReadFile(path.Join(cgroupCpuMountpoint, "cpu.cfs_period_us"))
+		_, err := ioutil.ReadFile(path.Join(cgroupCpuMountpoint, "cpu.shares"))
+		sysInfo.CpuShares = err == nil
+		if !sysInfo.CpuShares && !quiet {
+			logrus.Warn("Your kernel does not support cgroup cfs shares")
+		}
+		logrus.Warnf("sysInfo.CpuShares: lizf: %d", sysInfo.CpuShares)
```
Comment (hqhq) [inline pkg/sysinfo/sysinfo_linux.go]: Redundant warning needs to be removed.

**Comment by lizf-os** [comment]
@cpuguy83 now rebased, and I also sent out another PR to fix a sysinfo regression.

**Comment by jfrazelle** [comment]
I dont think this needs 3 commits can you please sqush to one

**Comment by icecrime** [comment]
Ping @lizf-os!

**Comment by LK4D4** [comment]
@lizf-os would you mind to rebase and squash commits?

**Comment by runcom** [comment]
Ping @lizf-os 
Let us know if you can handle this, I'm gonna carry it otherwise

**Comment by icecrime** [comment]
Ping @cpuguy83 @runcom: if one of you feels like carrying please do, I'm closing it.

**Comment by hqhq** [comment]
@icecrime @runcom After communicated with @lizf-os , I'm carrying this one, see #15320

**Comment by runcom** [comment]
thanks @hqhq

**Pull request closed without merging**